lib/blob.js

```diff
@@ -1216,14 +1215,141 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
 };
 
 /**
+ * Creates a new block or updates the content of an existing block blob.
+ * Updating an existing block blob overwrites any existing metadata on the blob,
+ * and the content of the existing blob is overwritten with the content of the new blob.
+ *
+ * @method putBlockBlob
+ * @param {string} container - Name of the container where the blob should be stored
+ * @param {string} blob - Name of the blob
+ * @param {object} options - Options on the following form
+ * ```js
+ * {
+ *    metadata: '...',                          // Name-value pairs associated with the blob as metadata
+ *    contentType: 'application/octet-stream',  // The MIME content type of the blob (optional)
+ *    contentEncoding: '...',                   // Specifies which content encodings have been applied to the blob. (optional)
+ *    contentLanguage: '...',                   // Specifies the natural languages used by this resource. (optional)
+ *    cacheControl: '...',                      // The Blob service stores this value but does not use or modify it. (optional)
+ *    disableContentMD5Check: 'false',          // Enable/disable the content md5 check is disabled.(optional)
+ *    leaseId: '...',                           // Lease id (required if the blob has an active lease)
+ *    contentDisposition: '...',                // Specifies the content disposition of the blob (optional)
+ *    ifModifiedSince: new Date(),              // Specify this to perform the operation only if the resource has been modified since the specified time.
+ *    ifUnmodifiedSince: new Date(),            // Specify this to perform the operation only if the resource has not been modified since the specified date/time.
+ *    ifMatch: '...',                           // ETag value. Specify this to perform the operation only if the resource's ETag matches the value specified.
+ *    ifNoneMatch: '...',                       // ETag value. Specify this to perform the operation only if the resource's ETag does not match the value specified.
+ * }
+ *```
+ * @param {string|buffer} content - The content of the blob
+ * @return {Promise} A promise for an object on the form:
+ * ```js
+ * {
+ *    eTag: '...',         // The entity tag of the blob
+ *    lastModified: '...', // The date/time the blob was last modified
+ *    contentMD5: '...',   // The MD5 hash of the blob
+ * }
+ * ```
+ */
+Blob.prototype.putBlockBlob = function putBlockBlob(container, blob, options, content){
```

Comment:
I am a fan of leaving the putBlob function in place as is, so that people can consume the API as specified in the Azure docs, also to make it easier to switch from `azure-storage` to `fast-azure-storage`.